### ISSUE TITLE
SyntaxWarning: "is not" with a literal

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2713]
With our recent Python version updates we are now seeing the following warnings when running our tests within an rpm build environment:
```
cd src/rockstor/
poetry run django-admin test
/src/rockstor/system/luks.py:241: SyntaxWarning: "is not" with a literal. Did you mean "!="?
if container_dev is not "":
/src/rockstor/system/luks.py:351: SyntaxWarning: "is not" with a literal. Did you mean "!="?
if len(source_dev_fields) is not 2:
/src/rockstor/storageadmin/views/email_client.py:204: SyntaxWarning: "is not" with a literal. Did you mean "!="?
if len(line) > 0 and line[0] is not "#":
/src/rockstor/storageadmin/views/rockon_helpers.py:239: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if new_state is not "pending_update" and task is not None:
/src/rockstor/storageadmin/views/rockon_helpers.py:239: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if new_state is not "pending_update" and task is not None:
/src/rockstor/storageadmin/views/rockon_helpers.py:379: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if c.uid is -1:
Creating test database for alias 'default'...
Creating test database for alias 'smart_manager'...
```

**Comment by phillxnet** [comment]
## New in Py3.8
https://docs.python.org/3.8/whatsnew/3.8.html#changes-in-python-behavior

> The compiler now produces a SyntaxWarning when identity checks (is and is not) are used with certain types of literals (e.g. strings, numbers). These can often work by accident in CPython, but are not guaranteed by the language spec. The warning advises users to use equality tests (== and !=) instead. (Contributed by Serhiy Storchaka in bpo-34850.)

**Comment by phillxnet** [comment]
When changing our second instance in a single line as recommended:

```
/src/rockstor/storageadmin/views/rockon_helpers.py:239: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if new_state is not "pending_update" and task is not None:
/src/rockstor/storageadmin/views/rockon_helpers.py:239: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if new_state is not "pending_update" and task is not None:
```

We run-into: 

> Comparison with None performed with equality operators 
> PEP 8: E711 comparison to None should be 'if cond is not None:'

Ignoring second instance for now for Draft PR preparation.

**Comment by phillxnet** [comment]
Double checking on the double "is not" use, and sure enough it seems to be a false alarm:
Carrying just one 'wrong' instance (as we currently have) invokes two warnings:

```
            if new_state is not "pending_update" and task is not None:
```
results in:
```
lbuildvm:/opt/rockstor/src/rockstor # poetry run python -Wd /opt/rockstor/.venv/bin/django-admin test -v 2
/opt/rockstor/.venv/lib/python3.9/site-packages/django/utils/version.py:98: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  loose_version = LooseVersion(version)
/opt/rockstor/src/rockstor/storageadmin/views/rockon_helpers.py:495: DeprecationWarning: invalid escape sequence \l
  [DOCKER, "exec", c.name, "psql", "-U", "postgres", "-c", "\l"],
/opt/rockstor/src/rockstor/storageadmin/views/rockon_helpers.py:239: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if new_state is not "pending_update" and task is not None:
/opt/rockstor/src/rockstor/storageadmin/views/rockon_helpers.py:239: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  if new_state is not "pending_update" and task is not None:
Creating test database for alias 'default' ('test_storageadmin')...
```
Where as with:
```
            if new_state != "pending_update" and task is not None:
```
gives the following:
```
lbuildvm:/opt/rockstor/src/rockstor # poetry run python -Wd /opt/rockstor/.venv/bin/django-admin test -v 2
/opt/rockstor/.venv/lib/python3.9/site-packages/django/utils/version.py:98: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  loose_version = LooseVersion(version)
/opt/rockstor/src/rockstor/storageadmin/views/rockon_helpers.py:495: DeprecationWarning: invalid escape sequence \l
  [DOCKER, "exec", c.name, "psql", "-U", "postgres", "-c", "\l"],
Creating test database for alias 'default' ('test_storageadmin')...
```
Moving to pull request.

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2722
